Sources/Segment/Utilities/Storage.swift

```diff
@@ -54,7 +58,11 @@ internal class Storage: Subscriber {
             default:
                 if isBasicType(value: value) {
                     // we can write it like normal
-                    userDefaults?.set(value, forKey: key.rawValue)
+                    let setUserDefaults = self.userDefaults?.safelySetUserDefaults(valueToWrite: value, keyToWrite: key.rawValue)
+                    
+                    if let error = setUserDefaults {
+                        analytics?.reportInternalError(error)
```

Comment:
Does `analytics.reportInternalError(error) handle that type of error?

Reply:
I think so because the type is the same I'm returning in `safelySetUserDefaults`. here's the method: 

```
    public func reportInternalError(_ error: Error, fatal: Bool = false) {
        let translatedError = Self.translate(error: error)
        configuration.values.errorHandler?(translatedError)
        Self.segmentLog(message: "An internal error occurred: \(translatedError)", kind: .error)
        if fatal {
            exceptionFailure("A critical error occurred: \(translatedError)")
        }
    }
   ```